Python/ast_opt.c

```diff
@@ -631,6 +632,63 @@ fold_compare(expr_ty node, PyArena *arena, _PyASTOptimizeState *state)
     return 1;
 }
 
+/* bpo-44501: Pack call arguments into a constant tuple if they are
+ *            all constants. The limits are important since the optimization
+ *            shows significance after 3 or more elements, and we don't want
+ *            to create giant tuples. */
+
+#define AST_MIN_CALL_FOLD 3
+#define AST_MAX_CALL_FOLD 100
+
+static int
+fold_call(expr_ty node, PyArena *arena, _PyASTOptimizeState *state)
+{
+    assert(node->kind == Call_kind);
+
+    asdl_expr_seq *args = node->v.Call.args;
+    Py_ssize_t nargs = asdl_seq_LEN(args);
+    Py_ssize_t nkeywords = asdl_seq_LEN(node->v.Call.keywords);
+
+    if (nkeywords || !nargs) {
```

Comment:
`!nargs` is redundant. We tests `nargs < AST_MIN_CALL_FOLD` below.

Reply:
Yes. But `AST_MIN_CALL_FOLD` can be `-1` (for example when someone else comes in and tinkers with these values), so we still need the bare minimum check of whether there are any positional arguments or not.